### ISSUE TITLE
[Bug] Clipboard copy from mobile to computer throws InvocationTargetException

### ISSUE THREAD

**Issue by Minabsapi** [Genymobile/scrcpy#1095]
**PC OS: Ubuntu 19.04**
**Android version: 8.1 (custom ROM: AEX)**

I get this error everytime I try to copy my phone clipboard to my PC's with `Ctrl + C` :
```
[server] ERROR: Could not invoke method
java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.genymobile.scrcpy.wrappers.ClipboardManager.getPrimaryClip(ClipboardManager.java:49)
	at com.genymobile.scrcpy.wrappers.ClipboardManager.getText(ClipboardManager.java:66)
	at com.genymobile.scrcpy.Device.getClipboardText(Device.java:149)
	at com.genymobile.scrcpy.Controller.handleEvent(Controller.java:100)
	at com.genymobile.scrcpy.Controller.control(Controller.java:67)
	at com.genymobile.scrcpy.Server$1.run(Server.java:47)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.SecurityException: Calling uid 0 does not own package com.android.shell
	at android.os.Parcel.readException(Parcel.java:2005)
	at android.os.Parcel.readException(Parcel.java:1951)
	at android.content.IClipboard$Stub$Proxy.getPrimaryClip(IClipboard.java:187)
	... 8 more
```
Any idea on what could be causing this issue ?

**Comment by rom1v** [comment]
What is the result of:

```
adb shell id
```
?

If you replace `0` by the uid given by the command here:

https://github.com/Genymobile/scrcpy/blob/31bd95022bc525be42ca273d59a3211d964d278b/server/src/main/java/com/genymobile/scrcpy/wrappers/ClipboardManager.java#L15

does this fix the problem?

**Comment by Minabsapi** [comment]
Just tested. Didn't have edit the USER_ID constant because the returned uid is 0.
It failed again

**Comment by rom1v** [comment]
Oh, on my device, the uid is 2000.

Could you paste the whole result of `adb shell id` please?

**Comment by Minabsapi** [comment]
Here is it: ``uid=0(root) gid=0(root) groups=0(root),1004(input),1007(log),1011(adb),1015(sdcard_rw),1028(sdcard_r),3001(net_bt_admin),3002(net_bt),3003(inet),3006(net_bw_stats),3009(readproc),3011(uhid) context=u:r:su:s0``

**Comment by rom1v** [comment]
Did you `adb root`? Or is it by default?

When you execute `adb shell`, the prompt is `#` instead of `$`, right?

**Comment by Minabsapi** [comment]
You are right, it seems the shell is root by default (`#` instead of `$` when doing `adb shell`)

**Comment by rom1v** [comment]
From your root shell, can you `su shell` or something like that?

From your computer, can you `adb shell su shell -c id`?

**Comment by rom1v** [comment]
What id you replace the `USER_ID`:

https://github.com/Genymobile/scrcpy/blob/31bd95022bc525be42ca273d59a3211d964d278b/server/src/main/java/com/genymobile/scrcpy/wrappers/ClipboardManager.java#L15

by the `shell` id, given by:
```
adb shell id shell
```

?

**Comment by Minabsapi** [comment]
Typing `su shell` gave me the following: `/system/bin/sh: /system/etc/mkshrc[30]: .: /sdcard/.aliases: Permission denied`
`adb shell id shell` gives `uid=2000(shell) gid=2000(shell) groups=2000(shell) context=u:r:magisk:s0`

Since I installed scrcpy through snap, I am unable to edit the files.

**Important note**
All these tests were made after upgrading Ubuntu to 19.10, which brought couple of issues:
* My device was not recognized through ADB at first, fixed after switching mode to FTP
* Once fixed, I tried the tests. shell is now default instead of root when started with `adb shell`
* I can no longer boot `scrcpy`. Log:
```
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
X Error:  GLXBadContext
  Request Major code 151 (GLX)
  Request Minor code 6 ()
  Error Serial #104
  Current Serial #103
```
(Will link to another issue)
